src/main/java/org/klodnicki/service/ReturnBookService.java

```diff
@@ -29,12 +36,34 @@ public void returnBook(String firstName, String lastName, String pesel, String t
         Account account = accountService.findAccountByFirstNameAndLastNameAndPesel(firstName, lastName, pesel);
         BookInfo bookInfo = bookService.findBookByTitleAndAuthorAndEdition(title, author, edition);
 
-        bookInfo.removeAccount(account);
+        LendingInformation lendingInformation = lendingInformationService.findLendingInformationByAccountAndBookInfo
+                (account, bookInfo);
+
+        if (countFineForKeepingBook(lendingInformation) > 0) {
+            System.out.println(FINE_TO_BE_PAID + countFineForKeepingBook(lendingInformation));
+        }
+
         int bookCopiesUpdate = bookInfo.getCopiesNumber() + 1;
         bookInfo.setCopiesNumber(bookCopiesUpdate);
 
         bookService.update(bookInfo);
+        lendingInformationService.remove(lendingInformation);
     }
 
+    private int countFineForKeepingBook(LendingInformation lendingInformation) {
 
+        LocalDateTime lendingDate = lendingInformation.getLendingDate();
+        LocalDateTime returningDate = LocalDateTime.now();
+
+        int daysOfYearWhenBookWasBorrowed = lendingDate.getDayOfYear();
```

Comment:
I found a bug. `getDayOfYear()` method returns a day of a year, as the name says. So in case we borrowed a book in December 2022 - let's say it was 360th day of a year - and then we return it on January 2023 - let's say it will be 3rd day of a year - the calculated `totalDays` will be negative number.

Also I'd like to point out naming like `daysOfYearWhenBookWasBorrowed`. This name is unnecessary long. In this case it could be `borrowDay` and code would look cleaner and sense would be kept.
Same goes for method name, could be just simple `countFine`. I'll not point out every single case of that, but please be aware of this.

With naming there is room for an improvement, keep working on it!

Reply:
You are right! I thought it will be counting as 365 + amount of days from a new year but it turned out it will not! OK I found other solution. Please check now.